src/Util/UrlBuilder.php

```diff
@@ -47,21 +47,77 @@ class UrlBuilder
 
     private string $baseUrl;
 
-    public function __construct(string $secret, string $baseUrl, HashingAlgorithm $hashingAlgorithm)
-    {
+    public function __construct(
+        #[\SensitiveParameter]
+        string $secret,
+        string $baseUrl,
+        HashingAlgorithm $hashingAlgorithm,
+    ) {
         $this->setSecret($secret);
         $this->setBaseUrl($baseUrl);
         $this->setHashingAlgorithm($hashingAlgorithm);
     }
 
+    /**
+     * Creates a new instance from environment variables.
+     *
+     * The optional parameters allow to specify some of the values, while the
+     * remaining values fall back to environment variables or to a default
+     * value.
+     *
+     * This method only exists to BC-support creating a BigBlueButton class
+     * without injecting the UrlBuilder instance.
+     *
+     * @internal
+     */
+    public static function fromEnvVars(
+        #[\SensitiveParameter]
+        ?string $secret = null,
+        ?string $baseUrl = null,
+        ?HashingAlgorithm $hashingAlgorithm = null,
+    ): static {
+        $secret ??= getenv('BBB_SECRET') ?: getenv('BBB_SECURITY_SALT');
+        if (false === $secret) {
+            throw new \RuntimeException("No BBB-Secret (or BBB-Salt) found! Please provide it either in constructor (2nd argument) or by environment variable 'BBB_SECRET' (or 'BBB_SECURITY_SALT')!");
+        }
+
+        $baseUrl ??= getenv('BBB_SERVER_BASE_URL');
+        if (false === $baseUrl) {
+            throw new \RuntimeException('No BBB-Server-Url found! Please provide it either in constructor '
+                . "(1st argument) or by environment variable 'BBB_SERVER_BASE_URL'!");
+        }
+
+        $hashingAlgorithm ??= HashingAlgorithm::SHA_256;
+
+        // Extending classes need to override this method, if they change the
+        // constructor signature.
+        // @phpstan-ignore new.static
+        return new static($secret, $baseUrl, $hashingAlgorithm);
+    }
```

Comment:
We need to decide whether the env vars should be still supported in the next major version.
Until that is decided, this static factory can remain `@internal`.